Linkta-org/core
# Feature/Enhance Data Structure For LinktaFlow & UserInput

## Description
Backend:
- [X] Simplify the LinktaFlow schema to include only essential fields & to optimize data structure
- [X] Remove redundant ID mapping
- [X] Relocate UserInput ID mapping from UserInputController to UserInputService - to adhere to the separation of concerns

Others
- [X] Update API design documentation
- [X] Re-organize type files - separate data model types into separate files

Todo
- [ ] Audit LinktaFlow zustand state on frontend
- [ ] Investigate why save LinktaFlow button not working
- [ ] Audit InputHistory data retrieval performance & decide whether to update the data models for User & UserInput

Fixes #328 #329

## Type of change

## How Can this be tested? Testing Plan to review this PR
1. Spin up app at root directory
2. Click a `userInput` on the side nav
3. Click the `TensorFlow dev tool icon` and you will see the following updated data structures:
4. LinktaFlow:
![image](https://github.com/user-attachments/assets/05bf575c-aae2-432a-9d8c-1944c3ae4f0b)
5. InputHistory:
![image](https://github.com/user-attachments/assets/1803d252-43ac-4e9e-8835-2e92d69a8dd7)

## Checklist:
